### DISCUSSION TITLE
How do I get current datetime in the SQLTemplate?

### DISCUSSION THREAD

**Discussion by liuyong888** [apache/superset#15664] [Q&A / Help] [answered]
@zhaoyongjie 
Hi
**Is your feature request related to a problem? Please describe.**
Superset 1.1.0
I want to get current datetime in SQL Template

**Additional context**
That seems like the wrong way to go.
It says datetime is Undefined.
![image](https://user-images.githubusercontent.com/45752479/125019981-90015280-e0aa-11eb-9303-6ffca38347ec.png)

**Comment by srinify** [comment]
Was this documentation useful at all?

https://superset.apache.org/docs/installation/sql-templating#available-macros

**Comment by zhaoyongjie** [comment, marked as answer]
we can make macros in the `config.py`, for instance:

```
from datetime import datetime

def current_datetime():
    return datetime.now().strftime("%Y-%m-%d")

JINJA_CONTEXT_ADDONS = {
    "current_datetime": current_datetime
}
```

ensure that the flag `ENABLE_TEMPLATE_PROCESSING` is enabled
```
FEATURE_FLAGS = {
    ...
    "ENABLE_TEMPLATE_PROCESSING": True,
}
```

Then use in SQLLab

```
select ...
from ...
where start > {{ current_datetime() }}
```

**Reply by Synarcs** [reply]
Hi, can  these functions added in Jinja context be used outside of sql lab in superset. For instance I want to use custom string transformation function or datetime transformation in an virtual dataset created by me where i can use this python functions to transform a column using sql in the calculated column feature of superset virtual dataset

**Reply by nytai** [reply]
Yes, you can use jinja templating in virtual datasets and and metrics and calculated columns, though and issue was recently raised around it not working for metrics defined on the dataset.
